edc-client/src/main/java/com/turn/edc/client/EDCClient.java

```diff
@@ -230,6 +230,26 @@ public void put(HostAndPort destination, String key, String subkey, byte[] value
 
 		router.store(new CacheInstance(destination), new StoreRequest(key, subkey, value, ttl));
 	}
+	
+	/**
+	 * Set the ttl for a key in the provided destination
+	 *
+	 * @param destination Destination host and port
+	 * @param key Top-level key
+	 * @param ttl TTL (in seconds) for the top-level key
+	 * 
+	 * @return whether setTTL operation fail or succeed
+	 *
+	 * @throws IOException 
+	 * @throws TimeoutException 
+	 * @throws InvalidParameterException 
+	 */
+	public boolean setTTL(HostAndPort hostAndPort, String key, int ttl)
+			throws TimeoutException, IOException, InvalidParameterException {
+		checkHostAndPort(hostAndPort);
+
+		return router.setTTL(new CacheInstance(hostAndPort, -1), key, ttl) ;
```

Comment:
Can you use the `CacheInstance()` constructor that only takes the `HostAndPort` object? I know I used this constructor in the `get()` method, but I shouldn't have haha